### ISSUE TITLE
Why L-BFGS performs different in tensorflow.compat.v1 and pytorch

### ISSUE THREAD

**Issue by gaohuizhang** [lululxvi/deepxde#1698]
Hi,

I tried to train the model with L-BFGS after Adam, but I got different results from tensorflow.compat.v1 and PyTorch, even if I use the exact same code, just a different backend. 
This is the loss history using tensorflow.compat.v1
![tensorflow](https://github.com/lululxvi/deepxde/assets/41465240/1b7f07a8-0682-4ce6-a920-427d85558cc0)
This is the loss history using Pytorch
![pytorch](https://github.com/lululxvi/deepxde/assets/41465240/f0383f25-1262-4062-8169-5d68afcc5542)

Does anyone have any idea why this happens? Does this mean we have to use TensorFlow if we want to use the L-BFGS optimizer?

Thanks a lot

**Comment by bakhtiyar-k** [comment]
Hello, try to use the same seed for both cases `dde.config.set_random_seed(1)`. Maybe the results are different due to different initialization

**Comment by gaohuizhang** [comment]
Thanks for your advice, but I tried several times and got similar results. I don't think it is caused by randomness of initialization.

**Comment by praksharma** [comment]
Because LBFGS is implemented differently in both libraries. You find the link the the particular implementation (papers) in the source code of tf and torch.
